### DISCUSSION TITLE
Enhanced Character Separated Values table format

### DISCUSSION THREAD

**Discussion by matthew-brett** [frictionlessdata/frictionlessdata.io#641] [General]
Just in case you hadn't seen it - this format seems minimal and well designed:

https://github.com/astropy/astropy-APEs/blob/master/APE6.rst

**Comment by rufuspollock** [comment]
@matthew-brett could you give a brief summary of the format here perchance? And good to hear from you 😄

**Comment by matthew-brett** [comment]
Quoting from [the spec](https://github.com/astropy/astropy-APEs/blob/master/APE6.rst):

> The proposed Enhanced CSV (ECSV) format has the following overall structure:
>
> * A header section which consists of lines that start with the # character and provide the table definition and data format via a YAML-encoded data structure.
> * An initial line in the header section which identifies the file as ECSV and provides a version number.
 > * A CSV-formatted data section in which the first line contains the column names and subsequent lines contains the data values.

See [header details](https://github.com/astropy/astropy-APEs/blob/master/APE6.rst#header-details) for details on the header.    There are standard fields in the header (below), but:

> Beyond the minimal standard, applications are free to create a custom data structure as needed using the top-level meta keyword. 

The single required keyword is `datatype`, which contains a list of column specifiers (see below).  Optional keywords are `delimiter`, `meta` and `schema`.

The column specifiers must give the column `name` and `datatype` and may also specify `unit`, `format`, `description` and `meta`, where `meta` is an arbitrary structure giving meta-data for the column. 

Currently the standard reader is in Astropy, but the project is currently discussing breaking it out into its own library - see [here](https://groups.google.com/forum/#!msg/astropy-dev/iOha8R59PX8/NzGn9_-rAQAJ).

And - hello from semi-sunny Birmingham ... :)
